platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/sync/config/SyncConfig.java

```diff
@@ -52,4 +53,5 @@ public record SyncConfig(
         @ConfigProperty(defaultValue = "true") boolean filterLikelyDuplicates,
         @ConfigProperty(defaultValue = "3s") Duration nonAncestorFilterThreshold,
         @ConfigProperty(defaultValue = "500ms") Duration syncKeepalivePeriod,
-        @ConfigProperty(defaultValue = "1m") Duration maxSyncTime) {}
+        @ConfigProperty(defaultValue = "1m") Duration maxSyncTime,
+        @ConfigProperty(defaultValue = "100") int maxSyncEventCount) {}
```

Comment:
I think we should start with a really high number, and then lower it when we are able to adequately test this

Reply:
I've disabled this feature by default (0 == no size limit). We can enable it when it gets to the perf environments where we can test it.